database/index.go

```diff
@@ -136,25 +183,88 @@ func (idx *Index) Delete(v document.Value, k []byte) error {
 	return engine.ErrKeyNotFound
 }
 
+// validatePivots returns an error when the pivots are unsuitable for the index:
+// - no pivots at all
+// - having pivots length superior to the index arity
+// - having the first pivot without a value when the subsequent ones do have values
```

Comment:
Actually, for n pivots, the n - 1 first must have a value I think.

```sql
CREATE INDEX idx_foo_a_b_c_d ON foo(a, b, c, d);

-- valid cases
WHERE a > 10 ;
WHERE a = 1 AND b > 10 ;
WHERE a = 1 AND b = 2 AND c > 10 ;
WHERE a = 1 AND b = 2 AND c = 3 AND d > 10 ;
```

Reply:
These rules are defining what valid pivots are from the perspective of the index, not from the query planner. 

The no pivots at all case is there to support the `StreamScanOperator` case where we have a max but no min value, which means the indexed will be traversed until the max is reached: 

```
CREATE INDEX idx_foo_a_b_c_d ON foo(a, b, c, d);

WHERE a < 10 ;
```

In that case, the ranges are: `Min: nil, Max: [10]` which will lead to iterate without pivots.